scenarios/bigquery.py

```diff
@@ -0,0 +1,121 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs a bigquery query, filters the results, and uploads the filtered and complete data to GCS."""
+
+import argparse
+import subprocess
+import sys
+import time
+import json
+
+DEFAULTBUCKET = "gs://k8s-metrics"
+DEFAULTPROJECT = "k8s-gubernator"
+
+def check(*cmd, **keyargs):
+    """Logs and runs the command, raising on errors."""
+    print >>sys.stderr, 'Run:', cmd
+    # If "stdin_string" keyword arg is present run command and communicate string to stdin
+    if "stdin_string" in keyargs:
+        in_string = keyargs["stdin_string"]
+        del keyargs["stdin_string"]
+        keyargs["stdin"] = subprocess.PIPE
+        proc = subprocess.Popen(*cmd, **keyargs)
+        proc.communicate(input=in_string)
+        return
+    subprocess.check_call(*cmd, **keyargs)
+
+def validate_metric_name(name):
+    """Validates that a string is useable as a metric name (Can be used as GCS object name)."""
+    if len(name) == 0:
+        raise ValueError("metric-name cannot be an empty string.")
+    if any(char in name for char in "\r\n[]*?#/\\"):
+        raise ValueError("metric-name cannot contain any line feeds, carriage returns, or the "
+                         "following characters: []*?#/\\")
+
+def do_query(query, out_filename, project):
+    """Executes a bigquery query, outputting the results to a file."""
+    with open(out_filename, "w") as out_file:
+        check(["bq", "query", "--format=prettyjson", "--project_id="+project], stdin_string=query,
+              stdout=out_file)
+
+def do_jq(jqfilter, data_filename, out_filename):
+    """Executes a jq command on a data file and outputs the results to a file."""
+    with open(out_filename, "w") as out_file:
+        check(["jq", jqfilter, data_filename], stdout=out_file)
+
+def run_metric(project, bucket_path, metric, query, jqfilter):
+    """Executes a query, filters the results, and uploads filtered and complete results to GCS."""
+    fulldata_filename = metric + time.strftime("-%Y-%m-%d.json")
+    latest_filename = metric + "-latest.json"
+
+    do_query(query, fulldata_filename, project)
+    do_jq(jqfilter, fulldata_filename, latest_filename)
+    check(["gsutil", "cp", fulldata_filename, latest_filename, bucket_path + metric + "/"])
+
+def main(configs, project, bucket_path):
+    """Loads metric config files and runs each metric."""
+    if len(project) == 0:
+        raise ValueError("project-id cannot be an empty string.")
+    if len(bucket_path) == 0:
```

Comment:
`if not bucket_path`

See the [style guide entry](https://google.github.io/styleguide/pyguide.html?showone=True/False_evaluations#True/False_evaluations).